src/main/java/org/snomed/snowstorm/extension/ExtensionAdditionalLanguageRefsetUpgradeService.java

```diff
@@ -147,37 +151,126 @@ private Map<Long, ReferenceSetMember> getReferencedComponents(String branchPath,
 		return result;
 	}
 
-	private List<ReferenceSetMember> addOrUpdateLanguageRefsetComponents(String branchPath, AdditionalRefsetExecutionConfig config, Map<Long, ReferenceSetMember> existing) {
+	private List<ReferenceSetMember> addOrUpdateLanguageRefsetComponents(String branchPath, AdditionalRefsetExecutionConfig config, Map<Long, ReferenceSetMember> languageRefsetMembersToCopy) {
 		BranchCriteria branchCriteria = versionControlHelper.getBranchCriteria(branchPath);
 		List<ReferenceSetMember> result = new ArrayList<>();
 		// batch here as the max allowed in terms query is 65536
-		for (List<Long> batch : partition(existing.keySet(), 10_000)) {
+
+		// Get all concepts that need changing from ToCopy members
+		Set<String> conceptIds = new HashSet<>();
+		for (List<Long> batch : partition(languageRefsetMembersToCopy.keySet(), 10_000)) {
+			NativeSearchQueryBuilder queryBuilder = new NativeSearchQueryBuilder()
+					.withQuery(boolQuery().must(branchCriteria.getEntityBranchCriteria(Description.class)))
+					.withFilter(termsQuery(DESCRIPTION_ID, batch))
+					.withPageable(LARGE_PAGE);
+
+			try (final SearchHitsIterator<Description> searchHitsIterator = elasticsearchTemplate.searchForStream(queryBuilder.build(), Description.class)) {
```

Comment:
@QuyenLy87 With updated query above we don't need to fetch concept ids using description ids.